src/builder.rs

```diff
@@ -0,0 +1,81 @@
+use std::collections::{BTreeSet, HashMap, HashSet};
+use std::iter::FromIterator;
+
+use curve25519_dalek_ng::scalar::Scalar;
+
+use crate::{AmountSecrets, Dbc, DbcContent, ReissueTransaction, Result};
+
+///! Unblinded data for creating to sn_dbc::DbcContent
+pub struct Output {
+    pub amount: u64,
+    pub owner: blsttc::PublicKey,
+}
+
+#[derive(Default)]
+pub struct TransactionBuilder {
+    pub inputs: HashMap<Dbc, AmountSecrets>,
+    pub outputs: Vec<Output>,
+}
+
+impl TransactionBuilder {
+    pub fn add_input(mut self, dbc: Dbc, amount_secrets: AmountSecrets) -> Self {
+        self.inputs.insert(dbc, amount_secrets);
+        self
+    }
+
+    pub fn add_inputs(mut self, inputs: impl IntoIterator<Item = (Dbc, AmountSecrets)>) -> Self {
+        self.inputs.extend(inputs);
+        self
+    }
+
+    pub fn add_output(mut self, output: Output) -> Self {
+        self.outputs.push(output);
+        self
+    }
+
+    pub fn add_outputs(mut self, outputs: impl IntoIterator<Item = Output>) -> Self {
+        self.outputs.extend(outputs);
+        self
+    }
+
+    pub fn build(self) -> Result<(ReissueTransaction, HashMap<crate::Hash, blsttc::PublicKey>)> {
```

Comment:
this is nice.  much cleaner for the caller.  I didn't like that client code had to calc and pass blinding_factor into DbcContent.

Reply:
Very nice.